src/dvclive/monitor_system.py

```diff
@@ -0,0 +1,158 @@
+import logging
+from typing import Dict, Union, Optional, List, Tuple
+from pathlib import Path
+
+import psutil
+from statistics import mean
+from threading import Event, Thread
+from funcy import merge_with
+
+from .error import InvalidDataTypeError
+
+
+logger = logging.getLogger("dvclive")
+MEGABYTES_DIVIDER = 1024.0**2
+GIGABYTES_DIVIDER = 1024.0**3
+
+MINIMUM_CPU_USAGE_TO_BE_ACTIVE = 20
+
+
+class _MonitorSystem:
+    """
+    Monitor system resources and log them to DVC Live.
+    Use a separate thread to call a `_get_metrics` function at  fix interval and
+    aggregate the results of this sampling using the average.
+    """
+
+    _plot_blacklist_prefix: Tuple = ()
+
+    def __init__(
+        self,
+        interval: float = 0.5,
+        num_samples: int = 10,
+        plot: bool = True,
+    ):
+        if not isinstance(interval, (int, float)):
+            raise InvalidDataTypeError("MonitorSystem.interval", type(interval))
```

Comment:
let's just use ValueError here? InvalidDataTypeError is primarily about user data point, etc ... not arguments

Reply:
the first argument here is usually the value that user passed, not name of the argument